ingest/anndata_.py

```diff
@@ -35,3 +44,66 @@ def validate(self):
         except ValueError:
             return False
 
+    @staticmethod
+    def generate_cluster_header(adata, clustering_name):
+        """
+        Based on clustering dimensions, write clustering NAME line to file
+        """
+        dim2 = ['NAME', 'X', 'Y']
+        dim3 = dim2 + ['Z']
+        clustering_dimension = adata.obsm[clustering_name].shape[1]
+        if clustering_dimension == 2:
+            headers = dim2
+        elif clustering_dimension == 3:
+            headers = dim3
```

Comment:
Non-blocking: This might be a simpler way:
```
dim = ['NAME', 'X', 'Y']
if clustering_dimension == 3:
    dim.append('Z')
elif clustering_dimension > 3:
    ...
```

Reply:
`2e07bee`